src/colmap/scene/reconstruction.cc

```diff
@@ -452,8 +452,8 @@ Reconstruction Reconstruction::Crop(
   for (const auto& image : images_) {
     auto new_image = image.second;
     new_image.SetRegistered(false);
-    for (auto& point2D : new_image.Points2D()) {
-      point2D.point3D_id = kInvalidPoint3DId;
+    for (point2D_t pid = 0; pid < new_image.NumPoints2D(); ++pid) {
```

Comment:
```suggestion
    const num_points2D = new_image.NumPoints2D();
    for (point2D_t point2D_idx = 0; point2D_idx < num_points2D; ++point2D_idx) {
```

Reply:
Added with type specifier in https://github.com/colmap/colmap/pull/2557/commits/6f716b8cdc107e076313934578a8a3eeeeab5ee3